presto-main/src/main/java/com/facebook/presto/operator/scalar/StringFunctions.java

```diff
@@ -448,6 +452,60 @@ public static Slice trim(@SqlType("varchar(x)") Slice slice)
         return SliceUtf8.trim(slice);
     }
 
+    @Description("remove the longest string containing only given characters from the beginning of a string")
+    @ScalarFunction("ltrim")
+    @LiteralParameters("x")
+    @SqlType("varchar(x)")
+    public static Slice leftTrim(@SqlType("varchar(x)") Slice slice, @SqlType(CodePointsType.NAME) int[] codePointsToTrim)
+    {
+        return SliceUtf8.leftTrim(slice, codePointsToTrim);
+    }
+
+    @Description("remove the longest string containing only given characters from the end of a string")
+    @ScalarFunction("rtrim")
+    @LiteralParameters("x")
+    @SqlType("varchar(x)")
+    public static Slice rightTrim(@SqlType("varchar(x)") Slice slice, @SqlType(CodePointsType.NAME) int[] codePointsToTrim)
+    {
+        return SliceUtf8.rightTrim(slice, codePointsToTrim);
+    }
+
+    @Description("remove the longest string containing only given characters from the beginning and end of a string")
+    @ScalarFunction("trim")
+    @LiteralParameters("x")
+    @SqlType("varchar(x)")
+    public static Slice trim(@SqlType("varchar(x)") Slice slice, @SqlType(CodePointsType.NAME) int[] codePointsToTrim)
+    {
+        return SliceUtf8.trim(slice, codePointsToTrim);
+    }
+
+    @ScalarOperator(OperatorType.CAST)
+    @SqlType(CodePointsType.NAME)
+    public static int[] castToCodePoints(@SqlType(StandardTypes.VARCHAR) Slice slice)
+    {
+        int[] codePoints = new int[safeCountCodePoints(slice)];
+        int position = 0;
+        for (int index = 0; index < codePoints.length; index++) {
+            codePoints[index] = getCodePointAt(slice, position);
+            position += lengthOfCodePoint(slice, position);
+        }
+        return codePoints;
+    }
+
+    private static int safeCountCodePoints(Slice slice)
+    {
+        int codePoints = 0;
+        for (int position = 0; position < slice.length(); ) {
+            int codePoint = tryGetCodePointAt(slice, position);
+            if (codePoint < 0) {
+                throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "Invalid UTF-8 encoding");
```

Comment:
`"Invalid UTF-8 encoding in characters to trim: " + slice.toStringUtf8()`. Change the error messages in tests accordingly. `toStringUtf8()` turns invalid characters into replacement characters.